client/meter/blockchain.js

```diff
@@ -58,21 +59,35 @@ async function myBuyerContracts() {
   return contracts;
 }
 
-async function availableContracts() {
-  const count = await EnergyMaster.contractCount();
-  const contracts = [];
-  for (let i = 0; i < count; i++) {
-    const contractEntity = await EnergyMaster.contracts(i);
-    const contractAddr = contractEntity[0];
-    const deregistered = contractEntity[2];
+async function availableContracts(req, res) {
+  const count = await EnergyMaster.contractCount()
+  const contracts = []
+
+  for (let i = req.offset; i < Math.min(req.offset + req.query.limit, count); i++) {
+    const contractEntity = await EnergyMaster.contracts(i)
+    const contractAddr = contractEntity[0]
+    const deregistered = contractEntity[2]
+
     if (!deregistered) {
-      const contract = makeEnergyContract(contractAddr);
-      const offeredAmount = await contract.offeredAmount();
-      const unitPrice = await contract.unitPrice();
+      const contract = makeEnergyContract(contractAddr)
+      const offeredAmount = await contract.offeredAmount()
+      const unitPrice = await contract.unitPrice()
+
       contracts.push({contractAddr: contractAddr, unitPrice: unitPrice, offeredAmount: offeredAmount})
+    } else {
+      // We always want to return req.limit items, so we increase the conditional of the loop
+      // if this contract is deregistered
+      req.offset++
     }
   }
-  return contracts;
+
+  pageCount = Math.ceil(count / req.query.limit)
```

Comment:
This should actually be registeredCount, but we don't have that functionality yet, hence I added it to Master contract (not yet tested).